### PR TITLE
ci: re-enables unit tests by marking broken tests and require all tests

### PR DESCRIPTION
**Pull request by terrykong**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

# Changelog 
- Please update the [CHANGELOG.md](/CHANGELOG.md) under next version with high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation? Make sure to also update the [NeMo Framework User Guide](https://docs.nvidia.com/nemo-framework/user-guide/latest/index.html) which contains the tutorials

# Checklist when contributing a new algorithm
- [ ] Does the trainer resume and restore model state all states?
- [ ] Does the trainer support all parallelism techniques(PP, TP, DP)?
- [ ] Does the trainer support `max_steps=-1` and `validation`?
- [ ] Does the trainer only call APIs defined in [alignable_interface.py](/nemo_aligner/models/alignable_interface.py)?
- [ ] Does the trainer have proper logging?

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -18,11 +18,35 @@
 def pytest_addoption(parser):
     """
     Additional command-line arguments passed to pytest.
-        --cpu: use CPU during testing (DEFAULT: GPU)
     """
     parser.addoption(
         "--cpu", action="store_true", help="pass that argument to use CPU during testing (DEFAULT: False = GPU)"
     )
+    parser.addoption("--flaky", action="store_true", default=False, help="Run flaky tests")
```
Comment (ko3n1g) [inline tests/conftest.py:25]: What the benefit of using this vs `pytest -m "flaky"` or `pytest -m "not flaky"`?

**Review by ko3n1g — COMMENTED** [review]
one question, but otherwise looks good! I like the matrix for functional tests. It will be easy to add selective triggering to it: We will compute the matrix in a `pre-flight` job before and add it json-encoded to the `Functional_Tests` job

File: tests/conftest.py (hunk 1 above)
Comment (ko3n1g) [reply tests/conftest.py:25]: having said that: Cool and useful learning about `parser.addoption`! Will probably steal some of this stuff for Mcore!

Comment (terrykong) [reply tests/conftest.py:25]: Nothing really, more of a convenience. One gap to this flag is you'd have to re-invent the more advanced conjunctive/disjunctive logic if someone wanted `pytest -m 'not flaky and (optimizer or checkpoint)'`.

Comment (terrykong) [reply tests/conftest.py]: i've removed the flag

**Review by ko3n1g — DISMISSED** [review]

File: tests/conftest.py [hunk 2]
```diff
@@ -18,13 +18,18 @@
...
         "--cpu", action="store_true", help="pass that argument to use CPU during testing (DEFAULT: False = GPU)"
     )
 
 
+def pytest_configure(config):
+    config.addinivalue_line(
+        "markers", "run_only_on(device): runs the test only on a given device [CPU | GPU]",
+    )
+
+
```
Comment (ko3n1g) [inline tests/conftest.py:32]: That's a nice trick to run `@pytest.mark.run_only_on("GPU")` by default while skipping `@pytest.mark.run_only_on("CPU")` by default.

I was thinking that `-m 'run_only_on("GPU") and not run_only_on("CPU")'` filters the same set of tests, but is more clunky than just `pytest ...`.

Cool!

**Review by ko3n1g — APPROVED** [review]

**Pull request merged**